### ISSUE TITLE
mrview: dark mode issues

### ISSUE THREAD

**Issue by maxpietsch** [MRtrix3/mrtrix3#1864]
Text in all `AdjustButton` boxes is not readable in all mrview tools in dark mode:
<img width="1073" alt="Screenshot 2020-01-05 13 26 14" src="https://user-images.githubusercontent.com/10046944/71780119-f5104f80-2fbe-11ea-952d-787e484acb3e.png">

If I take out the hard-coded background colour transition in the `AdjustButton` class:
```
--- a/src/gui/mrview/adjust_button.cpp
+++ b/src/gui/mrview/adjust_button.cpp
@@ -44,7 +44,7 @@ namespace MR
           setStyleSheet ((
               "QLineEdit { "
               "padding: 0.1em 20px 0.2em 0.3ex; "
-              "background: qlineargradient(x1:0, y1:0, x2:0, y2:0.2, stop:0 gray, stop:1 white) url(:/adjustbutton.svg); "
+              "background: url(:/adjustbutton.svg); "
```
it looks like this:

![image](https://user-images.githubusercontent.com/10046944/71780257-a2d02e00-2fc0-11ea-988b-1cf849c70d01.png)

Not exactly eye candy but at least readable.

Personally, I'd take the 3D effect out and emphasize the border:

```
+          const QColor bg (QWidget::palette().color(QWidget::backgroundRole()));
+          const QColor fg (QWidget::palette().color(QWidget::foregroundRole()));
+          const float r (0.4);
+          const QColor hl (fg.red() * (1.0-r) + bg.red() * r, fg.green() * (1.0-r) + bg.green() * r, fg.blue() * (1.0-r) + bg.blue() * r, 255);
           setStyleSheet ((
               "QLineEdit { "
               "padding: 0.1em 20px 0.2em 0.3ex; "
-              "background: qlineargradient(x1:0, y1:0, x2:0, y2:0.2, stop:0 gray, stop:1 white) url(:/adjustbutton.svg); "
+              "background: url(:/adjustbutton.svg); "
               "background-position: right; "
               "background-repeat: no-repeat; "
               "font-size: " + str(font().pointSize()) + "pt; "
-              "border: 1px solid grey; "
-              "border-color: black lightgray white gray; "
+              "border: 1px solid rgb("+str(hl.red())+","+str(hl.green())+","+str(hl.blue())+"); "
               "border-radius: 0.3em }").c_str());
```

<img width="698" alt="Screenshot 2020-01-05 14 32 33" src="https://user-images.githubusercontent.com/10046944/71780891-60125400-2fc8-11ea-8b2a-78f4a3f077c5.png">

![image](https://user-images.githubusercontent.com/10046944/71780898-7c15f580-2fc8-11ea-9ddc-3d28a22a0e7e.png)

Note that hard-coding colours does not allow changing the system colour scheme on the fly for existing mrview instances. As far as I can tell, all colours including text colours but not those of AdjustButton are updated making this text hard to read until mrview is restarted.

Any preferences? 

Also, @bjeurissen we should try to support or disable dark mode in the binary release: https://github.com/owncloud/client/commit/710a2af6b5496df4ee5cfe13a9c407cf27979de2

![image](https://user-images.githubusercontent.com/10046944/71780298-0fe3c380-2fc1-11ea-8bf0-20ecd2ff692f.png)

**Comment by Lestropie** [comment]
> Personally, I'd take the 3D effect out and emphasize the border:

:+1: Prioritise highlighting of user-manipulable elements over aesthetics, especially if the gymnastics don't give the desired result universally. Even with further manual manipulation it'd be tough to make it look like an outdent when both the button surface and the background are already so dark; and the effective *indent* effect in dark mode doesn't feel intuitive.

> Note that hard-coding colours does not allow changing the system colour scheme on the fly for existing mrview instances. 

:man_shrugging: Happy to swallow that limitation myself.

**Comment by maxpietsch** [comment]
Closed with #1865
